vignettes/profiling.Rmd

```diff
@@ -0,0 +1,68 @@
+---
+title: "Profiling Performance"
+author: "Thomas Lin Pedersen"
+date: "`r Sys.Date()`"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Profiling Performance}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+The performance of `gtable` is critical as it has the potential to slow down
+plot creation a lot. In order to continuously monitor the performance of gtable
+the following piece of code is used to generate a profiling and inspect it:
+
+```{r}
+library(ggplot2)
+library(profvis)
+
+p <- ggplot() + 
+  scale_x_continuous(limits = c(0, 1)) + 
+  scale_y_continuous(limits = c(0, 1))
+
+p_build <- ggplot_build(p)
+
+profile <- profvis(for (i in seq_len(100)) ggplot_gtable(p_build))
```

Comment:
Does repeating it 100x really help with prof vis?

Reply:
Yeah, it's too fast to yield any meaningful results with just a single iteration... I'm doing 1000 outside of the vignette to get more granularity